### DISCUSSION TITLE
Wondering how to run server commands from Buy Signs

### DISCUSSION THREAD

**Discussion by NosBos** [EssentialsX/Essentials#3865] [Q&A] [answered]
I am trying to set up a system on a server to allow people to purchase ranks which are organized by another plugin from the Buy Signs in Essentials. I thought this would be quite a simple alteration however, I cant seem to figure out where or if I can even directly print to the server console from the BuySign class.

Is this viable or is there a better way of achieving this functionality?

**Comment by triagonal** [comment, marked as answer]
While it would likely be possible to alter the Essentials signs code to do this, I doubt it would be a very easy or straightforward change. The code that handles sign interactions is many years old at this point and any extensions to it would become messy very quickly.

Instead, I would recommend using a dedicated plugin with economy support to run commands upon sign interactions like [CommandButtons](https://www.spigotmc.org/resources/commandbuttons-execute-commands-on-button-presses.60991/) (it works for signs too) or another "command signs" plugin. You should be able to find more options in the Spigot resources.

**Reply by NosBos** [reply]
Thanks!
